docs/FlaskSQLAlchemy.md

```diff
@@ -0,0 +1,419 @@
+        username = db.Column(db.String(80), unique=True, nullable=False)
+        email = db.Column(db.String(120), unique=True, nullable=False)
+
+        def __repr__(self):
+            username = f'User: {self.username}'
+            return username
+```
+**Note:** Save this as test.py file.Make sure to not call your application flask.py because this would conflict with Flask itself.
+
+## So what did the above code do?
+
+* First we imported the [Flask](https://flask.palletsprojects.com/en/1.1.x/api/#flask.Flask) class & then the [SQLAlchemy](https://flask-sqlalchemy.palletsprojects.com/en/2.x/api/#flask_sqlalchemy.SQLAlchemy) class.
+
+* Next we create an instance of this class. The first argument is the name of the application module or package. If you are using a single module (as in this example), you should use \__name__ because depending on if it’s started as application or imported as module the name will be different ('\__main__' versus the actual import name). This is needed so that Flask knows where to look for templates, static files, and so on.
+
+* Applications need some kind of configuration. There are different settings you might want to change depending on the application environment like toggling the debug mode, setting the secret key, and other such environment-specific things.
+  **config** behaves exactly like a regular dictionary but supports additional methods to load a config from files.
+
+* Connecting to a database: The **Engine** is the starting point for any SQLAlchemy application. It’s “home base” for the actual database and its DBAPI, delivered to the SQLAlchemy application through a connection pool and a Dialect. The general structure & more information can be found [here](https://docs.sqlalchemy.org/en/13/core/engines.html#database-urls). An Engine references both a Dialect and a Pool, which together interpret the DBAPI’s module functions as well as the behavior of the database.
+  * **Dialect** names include the identifying name of the SQLAlchemy dialect, a name such as sqlite, mysql, postgresql, oracle, or mysql. 
+  * A connection **pool** is a standard technique used to maintain long running connections in memory for efficient re-use, as well as to provide management for the total number of connections an application might use simultaneously.
+  * **Database Urls:** Usually can include username, password, hostname, database name as well as optional keyword arguments for additional configuration. In some cases a file path is accepted, and in others a “data source name” replaces the “host” and “database” portions. The typical form of a database URL is:
+    ```
+    dialect+driver://username:password@host:port/database
+    ``` 
+  * In the simplest form we can consider the above as **connection string** which tell us what kind of database we are talking to & how we should access it. For the above code snippet we have used this connection string
+    ```
+    sqlite:////tmp/test.db
```

Comment:
Here too

Reply:
As I used the above configuration that's why I explained it here as well